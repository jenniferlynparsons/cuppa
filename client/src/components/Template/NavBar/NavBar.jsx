import React from "react";
import { Link } from "react-router-dom";
import "./NavBar.scss";

class NavBar extends React.Component {
  render() {
    return (
      <nav
        data-testid="navbar"
        role="navigation"
        className="navbar is-dark"
        aria-label="main navigation"
      >
        <div className="navbar-brand">
          <Link
            className="navbar-item"
            to={this.props.auth.isAuthenticated ? "/tea-collection" : "/"}
          >
            Cuppa
          </Link>
        </div>
        <div className="navbar-menu is-active">
          <div className="navbar-start" />
          <div className="navbar-end">
            <Link className="navbar-item" to={"/tea-collection"}>
              Tea Collection
            </Link>
            <div className="navbar-item has-dropdown is-hoverable">
              <a href="/" className="navbar-link">
                My Account
              </a>

              <div className="navbar-dropdown">
                <Link className="navbar-item" to={"/dashboard"}>
                  Dashboard
                </Link>
                <Link className="navbar-item" to={"/tea-types"}>
                  My Tea Types
                </Link>
                <hr className="navbar-divider" />
                <span className="navbar-item">
                  <button
                    data-testid="logout"
                    className="button is-small is-dark"
                    onClick={this.props.handleLogoutClick}
                  >
                    Logout
                  </button>
                </span>
              </div>
            </div>
            <ul>
              <li>
                <Link
                  className={
                    this.props.auth.isAuthenticated
                      ? "button is-primary"
                      : "button is-disabled"
                  }
                  to={"/new-tea"}
                >
                  Add a Tea
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}

export default NavBar;
