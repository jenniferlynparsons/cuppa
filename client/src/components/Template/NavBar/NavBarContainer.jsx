import React, { Component } from "react";
import { connect } from "react-redux";
import { logoutUser } from "../../../actions/authActions";
import NavBar from "./NavBar";

class NavBarContainer extends Component {
  onLogoutClick = e => {
    e.preventDefault();
    this.props.logoutUser();
  };
  render() {
    return <NavBar auth={this.props.auth} onLogoutClick={this.onLogoutClick} />;
  }
}

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { logoutUser }
)(NavBarContainer);

export const NavBarContainerComponent = NavBarContainer;
