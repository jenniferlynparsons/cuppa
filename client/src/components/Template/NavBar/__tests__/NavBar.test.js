import React from "react";
import { create } from "react-test-renderer";
import NavBar from "../";

test("snapshot", () => {
  const c = create(<NavBar />);
  expect(c.toJSON()).toMatchSnapshot();
});
