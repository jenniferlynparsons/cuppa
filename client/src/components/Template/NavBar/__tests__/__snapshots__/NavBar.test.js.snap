// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-dark"
  role="navigation"
>
  <div
    className="navbar-brand"
  >
    <a
      aria-current="page"
      className="navbar-item"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Cuppa logo"
        src={Object {}}
      />
    </a>
  </div>
  <div
    className="navbar-menu is-active"
  >
    <div
      className="navbar-start"
    />
    <div
      className="navbar-end"
    >
      <ul>
        <li>
          <a
            className="button is-primary"
            href="/new-tea"
            onClick={[Function]}
          >
            Add a Tea
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
